sktime/transformers/series_as_features/hog1d.py

```diff
@@ -0,0 +1,195 @@
+import numpy as np
+import pandas as pd
+import numbers
+import math
+from sktime.utils.data_container import tabularize
+from sktime.utils.validation.series_as_features import check_X
+from sktime.transformers.series_as_features.base \
+    import BaseSeriesAsFeaturesTransformer
+
+
+"""
+The HOG1D Transformer proposed by:
+
+@article{zhao2015classifying,
+  title={Classifying time series using local descriptors with hybrid sampling},
+  author={Zhao, Jiaping and Itti, Laurent},
+  journal={IEEE Transactions on Knowledge and Data Engineering},
+  volume={28},
+  number={3},
+  pages={623--637},
+  year={2015},
+  publisher={IEEE}
+}
+"""
+
+
+class HOG1D(BaseSeriesAsFeaturesTransformer):
+
+    def __init__(self, num_intervals=2, num_bins=8, scaling_factor=0.1):
+        self.num_intervals = num_intervals
+        self.num_bins = num_bins
+        self.scaling_factor = scaling_factor
+        super(HOG1D, self).__init__()
+
+    """
+    Parameters
+    ----------
+    X : a pandas dataframe of shape = [n_samples, num_dims]
+        The training input samples.
+
+    Returns
+    -------
+    dims: a pandas data frame of shape = [n_samples, num_dims]
+    """
+    def transform(self, X, y=None):
+
+        # Check the data
+        self.check_is_fitted()
+        X = check_X(X, enforce_univariate=False)
+
+        # Get information about the dataframe
+        num_insts = X.shape[0]
+        col_names = X.columns
+
+        df = pd.DataFrame()
+
+        for x in col_names:
+            # Convert one of the columns in the dataframe to a numpy array
+            arr = tabularize(pd.DataFrame(X[x]), return_array=True)
+
+            # Get the HOG1Ds of each time series
+            transformedData = []
+            for y in range(num_insts):
+                inst = self.calculate_hog1ds(arr[y])
+                transformedData.append(inst)
+
+            # Convert to numpy array
+            transformedData = np.asarray(transformedData)
+
+            # Add it to the dataframe
+            colToAdd = []
+            for i in range(len(transformedData)):
+                inst = transformedData[i]
+                colToAdd.append(pd.Series(inst))
+
+            df[x] = colToAdd
+
+        return df
+
+    """
+    Function to calculate the HOG1Ds given a time series.
+
+    Parameters
+    ----------
+    X : a numpy array of shape = [num_atts]
+
+    Returns
+    -------
+    m : a numpy array of shape = [num_intervals*num_bins].
+        It contains the histogram of each gradient within
+        each interval.
+    """
+    def calculate_hog1ds(self, X):
```

Comment:
please also consider making the functions private here if not meant to be interacted with directly by users